### PR TITLE
Setuptools

### PR DESCRIPTION
**Pull request by stefoss23**
This PR changes sunbeam setup and compiling to using setuptools instead of cmake.

### CODE REVIEW

**Comment by stefoss23** [comment]
jenkins build this please

**Comment by stefoss23** [comment]
jenkins build this please

**Comment by stefoss23** [comment]
jenkins build this please

**Comment by stefoss23** [comment]
jenkins build this please

File: python/setup.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import sys
+import setuptools
+
+import glob
+import os
+
+ext_modules = [
+    Extension(
+        'libsunbeam',
+        glob.glob(os.path.join('src', '*.cpp')),
```
Comment (joakim-hove) [inline python/setup.py]: We generally do not use `glob` in opm - but rather list the source files explicitly.

**Comment by stefoss23** [comment]
jenkins build this please

**Comment by stefoss23** [comment]
jenkins build this please

File: CMakeLists.txt [hunk 2]
```diff
@@ -211,8 +211,48 @@ install(DIRECTORY cmake DESTINATION share/opm)
 install(FILES etc/opm_bash_completion.sh.in DESTINATION share/opm/etc)
 
 if (OPM_ENABLE_PYTHON)
+
+  find_package(PythonInterp REQUIRED)
+  include(FindPythonInterp)
+
+  if (EXISTS "/etc/debian_version")
```
Comment (joakim-hove) [inline CMakeLists.txt]: The lines 218 - 223 can be removed (I think)

File: CMakeLists.txt [hunk 3]
```diff
@@ -211,8 +211,48 @@ install(DIRECTORY cmake DESTINATION share/opm)
...
+
+  add_custom_command(TARGET sunbeam PRE_BUILD COMMAND python ${CMAKE_BINARY_DIR}/python/setup.py
+                                                      build_ext
+                                                      --build-lib=${CMAKE_BINARY_DIR}/python/python/sunbeam
+                                                      --library-dirs=${_setup_lib_dirs}
+                                                      --include-dirs=${_setup_include_dirs}
+                                                      WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/python
+                     )
+
+  INSTALL( DIRECTORY ${CMAKE_BINARY_DIR}/python/python/sunbeam/  DESTINATION ${CMAKE_INSTALL_PREFIX}/${PYTHON_INSTALL_PREFIX})
```
Comment (joakim-hove) [inline CMakeLists.txt]: The `INSTALL( ... )` should be removed; installation should be based on `setup.py install` - but that can wait.

File: python/setup.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import sys
+import setuptools
+
+import glob
+import os
+
+ext_modules = [
+    Extension(
+        'libsunbeam',
+        #glob.glob(os.path.join('src_sunbeam', '*.cpp')),
```
Comment (joakim-hove) [inline python/setup.py]: Just remove the commented out code.

**Comment by joakim-hove** [comment]
Instead of the name `src_sunbeam` I think the directory should be called e.g. `cxx`

**Comment by stefoss23** [comment]
jenkins build this please

File: python/setup.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    package_dir = {'': 'python'},
+    packages=[
+                'sunbeam',
+                'sunbeam.tools',
+                'sunbeam.deck',
+            ],
+    ext_modules=ext_modules,
+    license='Open Source',
+    zip_safe=False,
+    tests_suite=['tests'],
```
Comment (joakim-hove) [inline python/setup.py]: I think this should be: `test_suite = 'tests'` - at least I get a warning: 
```bash
/usr/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'tests_suite'
  warnings.warn(msg)
```
with the current

**Comment by joakim-hove** [comment]
In addition to being called from cmake - as is done in this PR, the `setup.py` based solution should also work in a 100% standalone mode where the library and headers from opm-common have already been installed. I have done the following:

1. Compile opm-common without Python and install to $OPM_PREFIX.
2. Wipe the `build/` directory
3. Go to: `opm-common/python` and run:
```bash
python setup.py build_ext --include-dirs=$OPM_PREFIX/include --library-dirs=$OPM_PREFIX/lib
```

That seems to work - very good! But we should verify that also the subcommands `build`, `test` and `install` work as expected - at least `build`does not seem to work yet.

File: python/setup.py (hunk 5 above)
Comment (joakim-hove) [reply python/setup.py]: NB - it should be `test_suite` - in *singular*

**Comment by stefoss23** [comment]
jenkins build this please

**Pull request merged**